### DISCUSSION TITLE
Document Template issues

### DISCUSSION THREAD

**Discussion by LyngeP** [elrumo/macOS_Big_Sur_icons_replacements#1384] [General]
I have previously used the Photoshop Production Templates found here: 
[https://developer.apple.com/design/resources/](https://developer.apple.com/design/resources/ ) to create App icons. 
Yesterday I wanted to make my first document file icon. 
But I have issues with the masks or the lack hereof so the generate option outputs images that doesn’t include the paper frame and fold or shadows.
Either Apple made a mistake with the template or I’m using it wrong, I hope it is the latter. 
Does anyone of you have experience with the Photoshop template?  And do you experience the same issue

**Comment by rodgerwerner** [comment]
The template you mention by Apple, I think is just a general idea of how to make them. Not, as one might think, to actually make the icons for your own use.  I believe this is because when a programmer uses Xcode to make filetype icons, much of the grunt or dirty work is done behind the scenes automatically. Even macOS now makes document icons behind the scenes for certain apps that do not include the image files themselves, but do include the specification for their app being associated with a particular extension. So, in these cases, macOS will automatically render the document, add the glyph icon of the app to the center of it, then add the extension text to the base of that icon file on the white page.

This is evidenced in the fact that all the gradients, page folds and masks are within a certain bundle file in the system library folder.

**Reply by LyngeP** [reply]
I get what you are saying, and I’m not saying you are wrong.
But there is something missed here then.
The template from apple is called: ”Production Templates” not ”sketch” or ”design” template.
And why would put an export function in the .psd file if it doesn’t exporting it correctly? not giving the designer a direct way of exporting the final .png’s just creates extra steps in several scenarios.  
I mean the template for the app icon works just fine and exports the icon with masks and everything.

And the setup you describe with logo and text from Xcode makes sense for that type of documents but not for documents with a full background image like the Xcode project document icon which I guess is rather hard to finetune from Xcode.

![2048](https://user-images.githubusercontent.com/7723958/174479363-6f7c2c5a-fa06-472e-9210-8eab1164f798.jpeg)

**Comment by rodgerwerner** [comment]
First off, don't get me wrong, I'm not saying Apple's document templates are perfectly fine ... clearly they are not.

The full image document icon is actually easier for Xcode to produce, because it is eliminating two steps ... remember, Xcode has the document and flaps images already with complimentary masking.

Also remember, although Apple does allow us to change the app icon for our own personal preference (in most cases), changing document icons is a bit 'hit and miss' with going into the resources folder, especially considering app designers can now sign every image in the included bundle...so that, when one changes any file, the entire app becomes 'corrupt'.

This is because Apple is trying to strive for a sense of visual continuity, hence the human interface guidelines ... I just wish Apple themselves would follow their HIG as even in the most recent OSes, there are still documents with the older curved flaps.
